lib/addons/fluxcd/gitrepository.ts

```diff
@@ -0,0 +1,30 @@
+import { FluxCDAddOnProps } from '.';
+import * as spi from "../../spi";
+
+/**
+ * Flux GitRepository API defines a Source to produce an Artifact for a Git repository revision.
+ */
+export class FluxGitRepository {
+
+    constructor(private readonly bootstrapRepo: spi.ApplicationRepository | undefined) {}
```

Comment:
why `| undefined`? It will never work you pass undefined and you check for null before calling. I suggest that you remove this.